src/index.ts

```diff
@@ -29,56 +29,62 @@ export * from './models'
  * @param {ApiOptions} options The API options object
  */
 export class Textile {
-  opts: ApiOptions
+  opts: ApiOptions = {
```

Comment:
ApiOptions is now an internal method because it always requires a url, port and version